test/Interop/Cxx/stdlib/overlay/custom-sequence-typechecker.swift

```diff
@@ -0,0 +1,49 @@
+// RUN: %target-typecheck-verify-swift -I %S/Inputs -enable-experimental-cxx-interop
```

Comment:
Can you make this an executable test?

Reply:
Yeah, I'll add an executable test to this PR once I get past the linker errors.

Reply:
The linker errors were an issue with my workflow, I was building `swiftstd-macosx-x86_64` instead of `swiftstd-macosx` 🙈

Added an executable test.